docs/input-logstash.asciidoc

```diff
@@ -107,7 +105,7 @@ This plugin supports the following configuration options plus the <<plugins-{typ
 |Setting                            |Input type        |Required
 | <<plugins-{type}s-{plugin}-host>> |<<string,string>> |No
 | <<plugins-{type}s-{plugin}-password>> |<<password,password>>|No
-| <<plugins-{type}s-{plugin}-port>> |<<number,number>> |Yes
+| <<plugins-{type}s-{plugin}-port>> |<<number,number>> |No
```

Comment:
lets remove this option.

Reply:
It is an downstream input (aka server) and it still needs to bind port of the given `host` network interface.
Upstream output (client) connects multiple (once we support HA) `hosts`. Anything am I missing? Didn't understand why we need to remove.

Reply:
Unclear why we should remove the configurability of the port? Without it, a Logstash instance could support at most one Logstash input plugin.

Eventually, I would like for Logstash itself to own the data path, and for these input/output plugins to be able to connect to a common port (maybe even the existing Logstash API?) and provide an address of sorts for each batch, so that pipelines on that downstream Logstash could effectively subscribe to the address and receive events sent to it. Creating a default port now before we are ready for that is a little counter-productive, because it prevents us from introducing a unified data path using the API's port later without breaking changes.

Reply:
yeah my bad, I thought this was the output.

Reply:
I loved the idea you brought @yaauie but some concerns I have:
- Logstash API can be also secured and customer should have an ability to provide credentials in the `logstash-output`; this will bring also another complexity;
- `logstash-input` doesn't have an ability to discover available port (if it fails on 9800, try 9801 or random) on the host for now. It means, the port is static 9800 by default unless customer intentionally changes; I mean by default port, it is a minimal change design now.
- I do believe this can be achieved by CPM for now;